### PR TITLE
(I#4539) Make side bar message dismissible

### LINKED ISSUE #4539: Make "Using facets and filters" dismissible
I'd like to be able to dismiss the "Using facets and filters" modal, which appears in the sidebar.

### Proposed solution
This would be a per-project setting; if a user dismisses the message for one project and starts a new one, the message would reappear on the new project. A small "x" in the left corner would suffice as an interface element, in my opinion.

### Alternatives considered
I realise the sidebar can be collapsed totally, but I don't think there are any alternatives to this specific problem.

### Additional context
Screenshot:
image

Version 3.5.2 [e3efd4e] on Safari Version 15.2 (17612.3.6.1.6) on macOS 12.1 (21C52).

Thanks,
Hugh

### PR DESCRIPTION
**Pull request by Babi-B**
Fixes #4539

Changes proposed in this pull request:
- Add a close icon button
- Message is dismissed when the close button is clicked
- Set a preference so users can prevent the sidebar message from appearing at all

### CODE REVIEW

File: main/webapp/modules/core/scripts/project/history-panel.html
Comment (wetneb) [inline main/webapp/modules/core/scripts/project/history-panel.html:6]: I wonder if we could not instead reuse an existing cross icon from the rest of the software. I would be surprised if we have not got any already. If we need to introduce a new one, I would avoid pasting its SVG code at two different locations in our source code.

Comment (Babi-B) [reply main/webapp/modules/core/scripts/project/history-panel.html:6]: I didn't find any cross image in the `images `directory (main/webapp/modules/core/images/). 

>I would avoid pasting its SVG code at two different locations in our source code.

I'll take out the redundancy

**Comment by Babi-B** [comment]
@wetneb I found a close.png image. Sorry for my hasty conclusion

**Comment by wetneb** [comment]
With my Firefox it looks like this:
image
There is quite a big empty margin at the top because of the button, that is perhaps not so elegant?

File: main/webapp/modules/core/styles/project/sidebar.less
Comment (antoine2711) [inline main/webapp/modules/core/styles/project/sidebar.less:115]:
```suggestion
  .close-btn:hover {
```

**Comment by Babi-B** [comment]
Oops! Let me correct that

**Comment by Babi-B** [comment]
@wetneb I was unable to re-create the image you had of your firefox browser display. On my browsers (Chrome, Firefox, Edge) the display was fine. I also checked it on different responsive screens using the inspection tool and it was fine.

I have changed the positioning a little. Could you please verify if it's fine now?

File: main/webapp/modules/core/scripts/project/history-panel.html
Comment (elebitzero) [reply main/webapp/modules/core/scripts/project/history-panel.html]: The jquery-ui icon set has an X icon.  I think it's: `<span class="ui-icon ui-icon-close">`

**Comment by elebitzero** [comment]
* Appropriate UI pattern would be X in the top left corner and/or button in the bottom right.
* Discard is not the right word.  I would use Close or Dismiss.
* With the current PR, you will need to dismiss the help panel each time you open the project. What the issue was asking for is a per-project setting that is saved. (makes more sense as a global user setting, not per project).   
* Optional: Could have a separate checkbox that indicates the choice should be saved. See the screenshot below from IntelliJ's Tip of the Day which shows a common pattern for this:

image

**Comment by antoine2711** [comment]
> * With the current PR, you will need to dismiss the help panel each time you open the project. What the issue was asking for is a per-project setting that is saved. (makes more sense as a global user setting, not per project).

I would agree with @elebitzero that this PR is not really solving the user problem as it would come back after each project opening.
Also, if we save this information in the metadata of the project, it's very hard to undo it for a user (though why would someone want to do that… ;-) )

A preference would address all these problems.

Regards, Antoine

**Comment by Babi-B** [comment]
Hello @antoine2711 

> A preference would address all these problems

I want to try setting the preference but I don't understand this line of code:

`JSON.parse(Refine.getPreference("ui.browsing.facetsHistoryPanelWidth", 300));`

 precisely `ui.browsing.facetsHistoryPanelWidth`. Where is this preference set? I've search through but found it mentioned only in the `docs` directory

**Comment by antoine2711** [comment]
> precisely `ui.browsing.facetsHistoryPanelWidth`. Where is this preference set? I've search through but found it mentioned only in the `docs` directory

@Babi-B: Go here: http://127.0.0.1:3333/preferences
image

`Refine.getPreference("ui.browsing.facetsHideSplash", false);` will get a preference value, if none exists, then the value is `false` (2nd argument).

Regards, Antoine

**Comment by Babi-B** [comment]
Oh, I get it now. Thank you @antoine2711 

Given this [comment](https://github.com/OpenRefine/OpenRefine/pull/4745#issuecomment-1098570933) @elebitzero made, I could add a checkbox with an option to set the preference to hide the sidebar message. So if the user checks the box the preference is set. The user could also set the preference manually if they wish.

Do you think this proposal is good or do you have another opinion?

File: main/webapp/modules/core/langs/translation-fr.json
Comment (antoine2711) [inline main/webapp/modules/core/langs/translation-fr.json]: Maybe just rename? `core-project/hide-panel`?

Comment (Babi-B) [reply main/webapp/modules/core/langs/translation-fr.json]: Ok but what about the value? I'm not sure my translation is right.

Comment (antoine2711) [reply main/webapp/modules/core/langs/translation-fr.json]: > Ok but what about the value? I'm not sure my translation is right.

I'm a native francophone, I can assure you. ;-) And Antonin is French! You are in good hands with us. ;-)

Regards, Antoine

Comment (Babi-B) [reply main/webapp/modules/core/langs/translation-fr.json]: Awesome!

**Comment by wetneb** [comment]
It looks like the Cypress test failure is not spurious: your changes probably throw some exception in some scenario (which you could find out from the CI logs).

File: main/webapp/modules/core/scripts/project/browsing-engine.js
Comment (antoine2711) [inline main/webapp/modules/core/scripts/project/browsing-engine.js:246]:
```suggestion
  var hideBrowsingFacetSplash = JSON.parse(Refine.getPreference("ui.browsing.hideFacetSplash", false));
```
No need for the Browsing to be repeated.

File: main/webapp/modules/core/scripts/project/browsing-engine.js
Comment (antoine2711) [inline main/webapp/modules/core/scripts/project/browsing-engine.js:245]: No need of Browsing is this function name. It's important for the prefs (category of pref) but not here where it's implied. Every thing here is abour browsing.

Comment (antoine2711) [inline main/webapp/modules/core/scripts/project/browsing-engine.js:245]:
```suggestion
BrowsingEngine.prototype.hideShowFacetSplash = function(clicked) {
```
No need of Browsing is this function name. It's important for the prefs (category of pref) but not here where it's in the parent object (BrowsingEngine). Every thing here is abour browsing.

File: main/webapp/modules/core/scripts/project/browsing-engine.js
Comment (antoine2711) [inline main/webapp/modules/core/scripts/project/browsing-engine.js:258]:
```suggestion
  } else {
    $('.browsing-panel-help').removeClass('hideFacetSplash');
  }
```
Just some spacing and indentation.

File: main/webapp/modules/core/scripts/project/browsing-engine.js
Comment (Babi-B) [reply main/webapp/modules/core/scripts/project/browsing-engine.js]: I created different preferences for the facet splash on the browsing div and the history div. If `hideBrowsingFacetSplash` is changed to `hideFacetSplash` it will cause both divs to be affected. Is that what is needed?

File: main/webapp/modules/core/scripts/project/history-panel.js
Comment (antoine2711) [inline main/webapp/modules/core/scripts/project/history-panel.js]:
```suggestion
  HistoryPanel.showHideSplash();
```
2 things here. 
1. History is not needed as it's in the parent name, and there is no link at all with facets.
2. The `HistoryPanel.prototype` syntax is necessary when creating/assigning the value, but not when calling the static function (as it is inherited). Just shorter and clearer.

File: main/webapp/modules/core/scripts/project/history-panel.js
Comment (antoine2711) [inline main/webapp/modules/core/scripts/project/history-panel.js]:
```suggestion
HistoryPanel.prototype.showHideSplash = function(clicked) {
```
As said earlier.

File: main/webapp/modules/core/scripts/project/history-panel.js
Comment (antoine2711) [inline main/webapp/modules/core/scripts/project/history-panel.js]:
```suggestion
  var hideHistorySplash = JSON.parse(Refine.getPreference("ui.browsing.hideHistoryFacetSplash", false));
```
I understand that you what to create ONE preference for the 2 splash, FacetSplash and HistorySplash? But it would be incoherent with the named used for the facetSplash… (`ui.browsing.hideBrowsingFacetSplash`).

You got to make a choice. ;-)

File: main/webapp/modules/core/styles/project/sidebar.less
Comment (antoine2711) [inline main/webapp/modules/core/styles/project/sidebar.less:153]: Shouldn't be another for the hideHistorySplash ?

Comment (Babi-B) [reply main/webapp/modules/core/styles/project/sidebar.less:153]: I didn't think it'll be necessary as the aim of that style is to set the splash display to none. I thought it'll be unnecessary repeition

**Review by antoine2711 — CHANGES_REQUESTED** [review]
You got a few mixed-up in the naming, and it does have implication in the code.

Let me know when you've fix that. Good work to bring the logic inside the `BrowsingEngine` and `HistoryPanel` objects.

Regards, Antoine

File: main/webapp/modules/core/styles/project/sidebar.less
Comment (antoine2711) [reply main/webapp/modules/core/styles/project/sidebar.less:153]: But don't you have 2 splashes? One for history and one for Facet? So, your intention are that when one user closes one, he closes the other also?! This is not clear to me yet.

Comment (Babi-B) [reply main/webapp/modules/core/styles/project/sidebar.less:153]: > So, your intention are that when one user closes one, he closes the other also?

Actually yes. Please correct me if my concept is wrong but I was thinking the user could want to close one splash but not the other. So I created two different preferences for the splashes. The `hideFacetSplash` class is reused in those different instances

Comment (antoine2711) [reply main/webapp/modules/core/styles/project/sidebar.less:153]: > Actually yes. Please correct me if my concept is wrong but I was thinking the user could want to close one splash but not the other.

K. So, then, this `.hideFacetSplash` could be called `.hideSplash`, `.hideLeftPanelSplash`, `.hideFacetHistorySplash`, but not just Facet, since it will be used in the 2.

File: main/webapp/modules/core/scripts/project/browsing-engine.js
Comment (antoine2711) [inline main/webapp/modules/core/scripts/project/browsing-engine.js:254]:
```suggestion
    $('.facet-splash-panel').addClass('hideLeftPanelSplash');
```
So, following the other comment, your styling is shared between the 2 spash, but the name must be unique and precise. `browsing-panel-help ` could be either, no?

Comment (antoine2711) [reply main/webapp/modules/core/scripts/project/browsing-engine.js:254]: Oups. I might have commented to fast… I see this: `history-panel-help`. This seams to be the original name (and it is precise).

File: main/webapp/modules/core/scripts/project/history-panel.js
Comment (Babi-B) [reply main/webapp/modules/core/scripts/project/history-panel.js]: @antoine2711: I just did a quick user test and `HistoryPanel.showHideSplash()` is not recognized as a function but `HistoryPanel.prototype.showHideHistoryFacetSplash()` is

File: main/webapp/modules/core/scripts/project/history-panel.html
Comment (antoine2711) [inline main/webapp/modules/core/scripts/project/history-panel.html:8]: Here, I don't see a common naming logic.
`close-side-panel-options` -> `panel-splash-close` ? Because it's shared amoung all the panels of type splash…
`side-panel-checkbox-label` -> `history-panel-checkbox-label` ? Side is very generic. It's an history panel from the start.
`close-btn` -> `panel-splash-close-btn` ? There could be other close button. Maybe the generic name is shared and though better. Just a thought.
`close-history-panel` -> `history-panel-close` ? Same as above. This is an ID, it must point uniquelly.

**Comment by Babi-B** [comment]
@antoine2711: I think the last 8 commits of this PR are, perhaps, irrelevant. Should I squash them or wait for all reviewers to review the work done first?

**Comment by antoine2711** [comment]
> @antoine2711: I think the last 8 commits of this PR are, perhaps, irrelevant. Should I squash them or wait for all reviewers to review the work done first?

Good question. Not sure of the perfect answer, but I would see that sqash are only important before merging in the master. Also, I think that you can merge only a range of commit, so that way you could not squash them all, by I really don't know, not being an expert in git… ;-) I would be safe, and wait before merge. But that just me. On my part, as long as you remember my comments, I don't care if you squash. I remember what I saw, if it's still there.

Regards, Antoine

**Comment by Babi-B** [comment]
Also, while working on OpenRefine/OpenRefine#4745 I came across a user manual where some preferences where listed out with a guide on they could be added. 

I suggest it should be modified if this PR is approved because the user might want to undo the `hideSplash` effect. There is no other way of doing that except through changing the preference

**Comment by antoine2711** [comment]
> Also, while working on #4745 I came across a user manual where some preferences where listed out with a guide on they could be added.
> 
> I suggest it should be modified if this PR is approved because the user might want to undo the `hideSplash` effect. There is no other way of doing that except through changing the preference

If you mean that we should also update the documentation in this PR so that the new preference appears there, you are quite correct. ;-)

Regards, Antoine

File: main/webapp/modules/core/scripts/preferences.js
Comment (wetneb) [inline main/webapp/modules/core/scripts/preferences.js:121]: Did we not already have those two functions somewhere else? If so, I would just like to be clear on why you needed to copy them over.

Comment (antoine2711) [reply main/webapp/modules/core/scripts/preferences.js:121]: > Did we not already have those two functions somewhere else? If so, I would just like to be clear on why you needed to copy them over.

@wetneb: right now, we do a small footrint of functions that are copied between the JS of pages: index.js, projects.js and preferences.js.

This like Refine.POSTcsrf(), etc. I'v written this function for v3.4 2 years ago. For the future, I'm looking at a small file to generalize them. Right now, this was helpful to code this issue. But it's really no big deal to keep it or delete it.

Regards, Antoine

File: main/webapp/modules/core/styles/project/sidebar.less
Comment (wetneb) [reply main/webapp/modules/core/styles/project/sidebar.less:153]: and maybe just `.hide-splash`, to keep a consistent style across class names?

File: main/webapp/modules/core/scripts/preferences.js
Comment (wetneb) [reply main/webapp/modules/core/scripts/preferences.js:121]: Ok but I do not get why you need to add those functions to `preferences.js` whereas the PR does not actually add any usage of those functions in this file - what did I miss?

Comment (antoine2711) [reply main/webapp/modules/core/scripts/preferences.js:121]: > Ok but I do not get why you need to add those functions to `preferences.js` whereas the PR does not actually add any usage of those functions in this file - what did I miss?

@wetneb : you were not ping on that discussion, but if you read [this](https://github.com/OpenRefine/OpenRefine/pull/4759#issuecomment-1099732745) (PR#4759), it will be clearer. It's a learning tool, and it's also code consolidation, factoring, at the cost of code already written and debugged.

Regards, Antoine

Comment (wetneb) [reply main/webapp/modules/core/scripts/preferences.js:121]: If those functions were added to `preferences.js` so that the developer could play with them via their Javascript console during development, I would remove them. Or did I still not get it?

Comment (antoine2711) [reply main/webapp/modules/core/scripts/preferences.js:121]: > If those functions were added to `preferences.js` so that the developer could play with them via their Javascript console during development, I would remove them. Or did I still not get it?

It's not just for that. I have code to use them IN the actual code, but I didn't want to mess this contributors stuff, just give a tool more. I plan to wait till 2 weeksm before submitting it. I somehow seam to think you already have enough on your plate. Right now, I just want to stay focus on Outreachy and help you. So if I can help anyway, you say.

Regards, Antoine

Comment (antoine2711) [reply main/webapp/modules/core/scripts/preferences.js:121]: @Babi-B: you can remove my code. And I will continue to review the rest of your work.

Regards, Antoine

File: main/webapp/modules/core/scripts/preferences.js
Comment (wetneb) [inline main/webapp/modules/core/scripts/preferences.js:35]: It looks like we can probably revert all changes in this file, so that this pull request does not touch this file at all, no?

File: main/webapp/modules/core/scripts/project/browsing-engine.js
Comment (antoine2711) [inline main/webapp/modules/core/scripts/project/browsing-engine.js:246]: @Babi-B: do you really need the `JSON.parse()` function?

File: main/webapp/modules/core/scripts/project/history-panel.html
Comment (antoine2711) [inline main/webapp/modules/core/scripts/project/history-panel.html:4]: @Babi-B: you added a new level of identation. It's not logic.

File: main/webapp/modules/core/scripts/project/history-panel.js
Comment (antoine2711) [inline main/webapp/modules/core/scripts/project/history-panel.js:78]: @Babi-B: now, you are no longer working in the facet panel, but rather the history panel. Your naming of function should show that.

File: main/webapp/modules/core/scripts/project/history-panel.js
Comment (antoine2711) [inline main/webapp/modules/core/scripts/project/history-panel.js:174]:
```suggestion
  var hideHistorySplash = JSON.parse(Refine.getPreference("ui.history.hideFacetSplash", false));
```
- Facet and History at the same time? 
- And, also, same for JSON as before
- The name of the preference is wrong if you want 2 separate prefs.

These have implication at other places. I won't specify them, please look for them.

**Review by antoine2711 — CHANGES_REQUESTED** [review]
You are getting there! Could you please also change the related documentation and add the 2 new preferences to the table of preferences?

https://github.com/OpenRefine/OpenRefine/blob/e582a7481e08eedbc9d0ae9419ec53cec7785e9d/docs/versioned_docs/version-3.5/manual/running.md#L333-L343

Regards, Antoine

File: docs/versioned_docs/version-3.5/manual/running.md
Comment (antoine2711) [inline docs/versioned_docs/version-3.5/manual/running.md:344]:
```suggestion
|Keep the history splash hidden|ui.history.hideFacetSplash|Boolean|false|true|—|
```
Is this an interesting change?

Comment (antoine2711) [reply docs/versioned_docs/version-3.5/manual/running.md]: @Babi-B: I've been very precise, but I really wasn't sure. Let's wait for someone else, maybe an english speaking, like @thadguidry or @ostephens. Also, if we change one, shouldn't we change the other?

Regards, Antoine

Comment (Babi-B) [reply docs/versioned_docs/version-3.5/manual/running.md]: @antoine: no I understand perfectly. I missed that when correcting my work

File: main/webapp/modules/core/scripts/project/history-panel.js
Comment (antoine2711) [inline main/webapp/modules/core/scripts/project/history-panel.js:179]: There is something weird here. You should just read the preference with the `Refine.getPreference()` once, at the construction time. After that, the DOM knows the value. Then, if the user changes the setting, then you write the preference, but still, you never have to read it back. It still current in the DOM.

I understand you code kinda works, but it's inefficient and openes the door to problems while complicating the logic, therefor the reading by others. Can you try to remodel it so it only calls `getPrefs()` once?

I haven't completely tested it, but the revision I got was not working on Chrome. Might be me, but... I'll check it back tomorrow.

Regards, Antoine

Comment (Babi-B) [reply main/webapp/modules/core/scripts/project/history-panel.js:179]: > You should just read the preference with the Refine.getPreference() once, at the construction time. After that, the DOM knows the value. Then, if the user changes the setting, then you write the preference, but still, you never have to read it back. It still current in the DOM.

@antoine: Sir I just need some clarification. 

In this function, `getPreference` is called just once. Do you mean that `Refine.getPreference()` should be called once for both this function and that for the history splash?

Comment (antoine2711) [reply main/webapp/modules/core/scripts/project/history-panel.js:179]: > > You should just read the preference with the Refine.getPreference() once, at the construction time. After that, the DOM knows the value. Then, if the user changes the setting, then you write the preference, but still, you never have to read it back. It still current in the DOM.
> 
> @antoine: Sir I just need some clarification.
> 
> In this function, `getPreference` is called just once. Do you mean that `Refine.getPreference()` should be called once for both this function and that for the history splash?

What I mean is that `getPreference()` is in a function that get called every time that a click is made, plus the fist time. So that means the actually call the `getPreference()` as many time as there are clicks, plus once for the initialisation. It should only be called ONCE per page loading. You can't have 2 authorities for the source of that, and once the page is loaded, the DOM owns the facts that the splash is displayed or visible. If that value changed behind the scene, there is no purpose to reflect it in the interface. At that stage, the user controls the interface and the display of the splashes.

And no, I see no need/advantage to manage Facet & History splash in the same code. Some function can be written in a generic way to be used by both splashes (or even more places), but then, it's because they are generic.

Regards, Antoine

File: docs/versioned_docs/version-3.5/manual/running.md
Comment (antoine2711) [inline docs/versioned_docs/version-3.5/manual/running.md:344]:
```suggestion
|Keep the facet's splash hidden|ui.browsing.hideFacetSplash|Boolean|false|true|—|
|Keep the history's splash hidden|ui.history.hideHistorySplash|Boolean|false|true|—|
```

I've just seen this: `ui.browsing` must stay consistent. I understand why you went for that (other naming in the code structure), but at the user level, all of this is browsing, one is about the facet panel and the other about the history panel, both of which are in the `leftPanel`.

So, the logic of name is: `ui.browsing.hide{panelName}Splash`, where `panelName` = `Facet` of `History`.

Regards, Antoine

**Review by antoine2711 — CHANGES_REQUESTED** [review]
Hi @Babi-B, I fear we have taken a step behind. The checkbox is not working…
image

Also, in the naming, there is a new problem.

Regards, Antoine

File: docs/versioned_docs/version-3.5/manual/running.md
Comment (Babi-B) [reply docs/versioned_docs/version-3.5/manual/running.md:344]: @antoine2711: Okay

**Comment by Babi-B** [comment]
> Also, in the naming, there is a new problem.

@antoine2711 : I have updated the preference name. Is it ok now?

> The checkbox is not working…

Sir could you check now? It's working fine on my end

File: main/webapp/modules/core/scripts/project/history-panel.js
Comment (Babi-B) [reply main/webapp/modules/core/scripts/project/history-panel.js]: @antoine2711 : Please give me a second to check this

**Comment by antoine2711** [comment]
> Sir could you check now? It's working fine on my end

Well, @Babi-B, on my side, closing of the splashes works when click the « delete » button. What is not working is the saving of preferences.

image

Here, I put a breakpoint at the first instruction in the `setPreferences()` function. And it never stops there, meaning this code is never called. Also, the other test shown earlier is clear, the preference is still not saved.

Here, the console is really the most powerfull debugging tool for front-end javascript. My two tests should always work.

Regards, Antoine

**Comment by Babi-B** [comment]
@antoine2711: Sir please be patient with me. I honestly I'm unable to recreate what you have

> Here, I put a breakpoint at the first instruction in the setPreferences() function. And it never stops there, meaning this code is never called. Also, the other test shown earlier is clear, the preference is still not saved.

I am not very experienced with using the browser debugging tool. However, I added a `console.log` to the `Refine.setPreference` function and it is called when I check the checkbox to set the preference. Twice for each splash. Also the splash does not display when the preference is set.

image

Could you perhabs guide my on how to recreate what you have?

**Comment by antoine2711** [comment]
> Could you perhaps guide me on how to recreate what you have?

K. I'll check again on my side. No problem.

Regards, Antoine

**Comment by antoine2711** [comment]
> @antoine2711: Sir please be patient with me. I, honestly, am unable to recreate what you have

You don't have to ask me for patience. I always have patience for people in a sincere learning process. ;-)

> I am not very experienced with using the browser debugging tool. However, I added a `console.log` to the `Refine.setPreference` function in the `main/webapp/modules/core/scripts/project.js` file and it is called when I check the checkbox to set the preference. Once for each splash. Also, the splash does not display when the preference is set.

Your coding skills are good, and it's not what is in question here. It's the understanding of the user logic that's failing. Let me synthetize it.

- When the user click the Delete (or it probably should be Close) button, it should just close the splash panel.
- When the user click the checkbox, the preference `ui.browsing.hide{Panel}Splash` should be set to true.

Now, you do BOTH on the button click, and nothing on the checkbox click…
It is clearer?

Regards, Antoine

**Comment by Babi-B** [comment]
>You don't have to ask me for patience. I always have patience for people in a sincere learning process. ;-)

Thank you sir

>Now, you do BOTH on the button click, and nothing on the checkbox click…
It is clearer?

Actually it works as you previously described. I have check and it doesn't have this error.

**Comment by antoine2711** [comment]
> Actually it works as you previously described. I have checked and it doesn't have this error.

Let me summarize the logic:
- The user wants to hide the splash; he asked for a close button.
- In this issue, he ask that it be on a project basis; but we can't.
- We propose instead a preference, it's across a user, will still answer the user request
- To set the preference, we added a checkbox.

So, in conclusion, the close (or delete) button hide the splash, but the preference makes this closing permanent.
Now, if you check the checkbox, no preferences are set, but if you close the splash, a preference gets set.

Are we on the same page?

Regards, Antoine

**Comment by Babi-B** [comment]
@antoine2711: Let me just show you a video demo of the work. I have done two recordings of one splash because one is too large and GitHub does not support that.

In the first video when just the delete button is clicked

https://user-images.githubusercontent.com/66228179/165709184-d45f1b39-011a-48ca-83fe-d3a594ffb632.mp4

In this second video, the checkbox is checked.

https://user-images.githubusercontent.com/66228179/165709579-b98f69c1-63e1-40ba-94b9-548ecc28867d.mp4


As long as the preference is set to `true`, the splash does not display

**Comment by antoine2711** [comment]
> @antoine2711: Let me just show you a video demo of the work.

K. I'll check it again. Regards, Antoine

**Comment by antoine2711** [comment]
@Babi-B: sorry for the delay, I was very busy.

So, I understand that it works the way you use it. But there are logical user interface flaws in its implementation. The preference must be set when you check the checkbox, not after checking the checkbox AND pressing the close button.

Also, I see a little problem related to this implementation, and it's with the permalink. It should be updated as to not show the splash if it's click and the preference is set. I you analyse the permalink, it gives this:
'http://127.0.0.1:3333/project?project=2355996207070&ui={"facets"%3A[]}'
And this force the splash to be displayed, even if the preference for the splash is set to hide.

If you want a videocall with me, we could arrange that.

Regards, Antoine

**Review by antoine2711 — CHANGES_REQUESTED** [review]
I see 2 problems:
1. the preference should be saved strait when the checkbox is clicked,
2. if you open a project with the permalink, the splash is always shown, even if the hideSplash preference is set.

Regards, Antoine

**Comment by Babi-B** [comment]
Hello @antoine2711 

>the preference should be saved strait when the checkbox is click

I have a question about this. As a user, I could check the checkbox then change my mind and undo it before clicking the delete button.

So should I put a listening event on the checkbox so on change a function could set or remove the preference?

**Comment by wetneb** [comment]
Closing per inactivity, feel free to reopen if there is some movement.

**Pull request closed without merging**
